Assets/Editor/CreatureHydrator.cs

```diff
@@ -225,11 +225,32 @@ private static void HydrateCreature(UnitDefinition unit, string generaName, stri
         // Load the newly generated script
         MonoScript newScript = AssetDatabase.LoadAssetAtPath<MonoScript>(scriptPath);
         
-        // Attach the script to the prefab
+        // Attach the script to the prefab as a component
         Type scriptType = newScript.GetClass();
         if (scriptType != null)
         {
-            unitPrefab.AddComponent(scriptType);
+            // Add the script component to the prefab
+            var scriptComponent = unitPrefab.AddComponent(scriptType);
+
+            // Set the Animator field
+            var animatorField = scriptType.GetProperty("Animator");
+            if (animatorField != null && animatorField.CanWrite)
```

Comment:
nit, helper method for setting these properties?

Reply:
good point not doing it lawl